pypeit/core/wavecal/echelle.py

```diff
@@ -0,0 +1,217 @@
+    """
+    Predict an echelle spectrum wavelength solution for each order by evluating the polynomial fits of
+    wavelength solution coefficients vs echelle angle at the given echelle angle.
+
+    Args:
+        angle_fits_params (astropy.table.Table):
+            Table holding the parameters governing the echelle angle fits
+        ech_angle_coeffs (numpy.ndarray):
+            Array holding the polynomial coefficients for the fits of the wavelength solution polynomial coefficients
+            vs echelle angle.
+        ech_angle (float):
+            Echelle angle
+        order_vec (numpy.ndarray):
+            Array of order numbers for the deisred predicted spectrum. Shape = (norders,)
+        nspec (int):
+            Number of spectral pixels in the echelle spectrum
+
+    Returns:
+        wave_soln_guess (numpy.ndarray):
+            Array containing the predicted echelle spectrum. Shape = (nspec, norders)
+
+    """
+
+    norders = order_vec.size
+    wave_soln_guess = np.zeros((nspec, norders))
+
+    xnspecmin1 = float(nspec - 1)
+    xnspec = np.arange(nspec)/xnspecmin1
+
+    for iord, order in enumerate(order_vec):
+        # Index of the order in the total order vector used cataloguing the fits in the coeff arxiv
+        indx = order - angle_fits_params['order_min']
+        coeff_predict = np.zeros(angle_fits_params['ech_n_final'] + 1)
+        # Evaluate the coefficients for this order and the current ech_angle
+        for ic in range(angle_fits_params['ech_n_final'] + 1):
+            coeff_predict[ic] = fitting.evaluate_fit(
+                ech_angle_coeffs[indx, ic, :], angle_fits_params['ech_func'],
+                ech_angle, minx=angle_fits_params['ech_xmin'], maxx=angle_fits_params['ech_xmax'])
+
+        wave_soln_guess[:, iord] = fitting.evaluate_fit(coeff_predict, angle_fits_params['wave_func'], xnspec,
+        minx=angle_fits_params['wave_xmin'], maxx=angle_fits_params['wave_xmax'])
+
+
+    return wave_soln_guess
+
+
+def predict_ech_arcspec(angle_fits_file, composite_arc_file, echangle, xdangle, xdisp, nspec, norders, pad=3):
+    """
+    Predict the echelle arc spectrum using the fits to wavelength solution vs echangle and xdangle  and the archived
+    composite arcs.
+    Args:
+        angle_fits_file (str):
+            File containing the fits to wavelength solution vs echangle and xdangle
+        composite_arc_file (str):
+            File containing the archived composite arcs for each order.
+        echangle (float):
+            Echelle angle
+        xdangle (float):
+            Cross-disperser angle
+        xdisp (str):
+            Cross disperser. E.g. for Keck HIRES this is either 'UV' or 'RED'
+        nspec (int):
+            Number of spectral pixels in the echelle spectrum
+        norders (int):
+            Number of orders in the echelle spectrum
+        pad (int):
+            Number of orders to pad the coverage by on the blue and red side.
+    Returns:
+
```

Comment:
What is returned?  And you likely need a blank line above `Returns:`

Reply:
Added.